### PR TITLE
PWA-956 optimize pdp reviews

### PR DESCRIPTION
**Pull request by devbucket**

### CODE REVIEW

File: components/Reviews/index.jsx [hunk 1]
```diff
@@ -1,42 +1,44 @@
...
-  );
-};
+const Reviews = ({ productId }) => (
+  <Fragment>
+    <Portal name={PRODUCT_REVIEWS_BEFORE} />
+    <Portal name={PRODUCT_REVIEWS}>
+      {appConfig.hasReviews && (
+        <div className={styles.container} data-test-id="reviewSection">
+          <Header productId={productId} />
+          <List productId={productId} />
```
Comment (fkloes) [inline components/Reviews/index.jsx]: The `List` doesn't show reviews when it's used within the `Reviews` page

File: components/Reviews/index.jsx [hunk 2]
```diff
@@ -1,42 +1,44 @@
-import React from 'react';
+import React, { Fragment } from 'react';
```
Comment (fkloes) [inline components/Reviews/index.jsx]: When a `Reviews` page is reloaded, the `Header` and the `List` don't show data

File: components/Reviews/components/AllReviewsLink/connector.js [hunk 3]
```diff
@@ -8,8 +7,20 @@ import { getProductReviewCount } from '@shopgate/pwa-common-commerce/reviews/sel
...
 });
 
-export default connect(mapStateToProps);
+/**
+ * @param {Object} next The next component props.
+ * @param {Object} prev The previous component props.
+ * @returns {boolean}
+ */
+const areStatePropsEqual = (next, prev) => {
+  if (prev.totalReviewCount !== next.totalReviewCount) {
```
[the remark this answers is not in the document]
Comment (devbucket) [reply components/Reviews/components/AllReviewsLink/connector.js]: Done

File: components/Reviews/components/Header/index.jsx [hunk 4]
```diff
@@ -1,11 +1,7 @@
-import React, { Fragment } from 'react';
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (devbucket) [reply components/Reviews/components/Header/index.jsx]: Fixed

File: components/Reviews/index.jsx (hunk 1 above)
Comment (devbucket) [reply components/Reviews/index.jsx]: Done

File: components/Reviews/index.jsx (hunk 2 above)
Comment (devbucket) [reply components/Reviews/index.jsx]: Done

**Pull request merged**